powershell/VstsTaskSdk/ServerOMFunctions.ps1

```diff
@@ -48,10 +48,145 @@ $projectHttpClient.GetProjects().Result
 
-    Add-Type -LiteralPath (Assert-Path "$(Get-TaskVariable -Name 'Agent.ServerOMDirectory' -Require)\Microsoft.VisualStudio.Services.Common.dll" -PassThru)
-    $endpoint = (Get-Endpoint -Name SystemVssConnection -Require)
-    New-Object Microsoft.VisualStudio.Services.Common.VssServiceIdentityCredential(
-        New-Object Microsoft.VisualStudio.Services.Common.VssServiceIdentityToken([string]$endpoint.auth.parameters.AccessToken))
+    Trace-EnteringInvocation -InvocationInfo $MyInvocation
+    try {
+        if (!$OMDirectory) {
+            # Fallback to the directory containing the entry script.
+            $OMDirectory = [System.IO.Path]::GetFullPath("$PSScriptRoot\..\..")
+        }
+
+        # Get the credentials.
+        $endpoint = (Get-Endpoint -Name SystemVssConnection -Require)
+        Add-Type -LiteralPath (Assert-Path "$(Get-TaskVariable -Name 'Agent.ServerOMDirectory' -Require)\Microsoft.VisualStudio.Services.Common.dll" -PassThru)
+        Add-Type -LiteralPath (Assert-Path "$(Get-TaskVariable -Name 'Agent.ServerOMDirectory' -Require)\Microsoft.VisualStudio.Services.WebApi.dll" -PassThru)
+        # Add-Type -LiteralPath (Assert-Path "$(Get-TaskVariable -Name 'Agent.ServerOMDirectory' -Require)\Microsoft.VisualStudio.Services.Client.dll" -PassThru)
+        New-Object Microsoft.VisualStudio.Services.Common.VssCredentials(
+            (New-Object Microsoft.VisualStudio.Services.Common.WindowsCredential($false)), # Do not use default credentials.
+            (New-Object Microsoft.VisualStudio.Services.OAuth.VssOAuthAccessTokenCredential($endpoint.auth.parameters.AccessToken)),
+            [Microsoft.VisualStudio.Services.Common.CredentialPromptType]::DoNotPrompt)
+
+        # Add-Type -LiteralPath (Assert-Path "$(Get-TaskVariable -Name 'Agent.ServerOMDirectory' -Require)\Microsoft.VisualStudio.Services.Common.dll" -PassThru)
+        # $endpoint = (Get-Endpoint -Name SystemVssConnection -Require)
+        # New-Object Microsoft.VisualStudio.Services.Common.VssServiceIdentityCredential(
+        #     New-Object Microsoft.VisualStudio.Services.Common.VssServiceIdentityToken([string]$endpoint.auth.parameters.AccessToken))
+    } finally {
+        Trace-LeavingInvocation -InvocationInfo $MyInvocation
+    }
+}
+
+function New-VssHttpClient {
+    [CmdletBinding()]
+    param(
+        [string]$OMDirectory,
+
+        [Parameter(Mandatory = $true)]
+        [string]$TypeName,
+
+        [string]$Uri,
+
+        $VssCredentials)
+
+    Trace-EnteringInvocation -InvocationInfo $MyInvocation
+    try {
+        if (!$Uri) {
+            # Default the URI.
+            $Uri = Get-TaskVariable -Name System.TeamFoundationCollectionUri -Require
+        }
+
+        [uri]$Uri = New-Object System.Uri($Uri)
+
+        if (!$VssCredentials) {
+            # Default the credentials.
+            $VssCredentials = Get-VssCredentials
+        }
+
+        # Try to load the type.
+        try {
+            $null = [type]$TypeName
```

Comment:
You know too much PS. I think $TypeName -is [Type] would be somewhat more readable.